src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java

```diff
@@ -276,35 +275,36 @@ public void onColonyTick(@NotNull final IColony colony)
     @Override
     public boolean canRecipeBeAdded(final IToken token)
     {
-        if (!super.canRecipeBeAdded(token) || !AbstractBuildingCrafter.canBuildingCanLearnMoreRecipes(getBuildingLevel(), super.getRecipes().size()))
-        {
-            return false;
-        }
 
-        final IRecipeStorage storage = IColonyManager.getInstance().getRecipeManager().getRecipes().get(token);
-        if (storage == null)
-        {
+        Optional<Boolean> isRecipeAllowed;
+
+        if (!super.canRecipeBeAdded(token)) {
             return false;
         }
 
-        for (final IRecipeStorage recipe : BakerRecipes.getRecipes())
-        {
-            if (recipe.getPrimaryOutput().isItemEqual(storage.getPrimaryOutput()))
-            {
-                return false;
-            }
-        }
+        isRecipeAllowed = super.canRecipeBeAddedBasedOnTags(token);
+        if (isRecipeAllowed.isPresent()) {
```

Comment:
the format is messed up in all these classes though.

Reply:
What do you mean?

Reply:
lala {

}

and not 

lala 
{

}

Reply:
Oh, is that "messed" up? :)

Reply:
Compared to the default formatting yeah =D

Reply:
Looks like this was the only one